app/models/target.rb

```diff
@@ -25,4 +25,16 @@ class Target < ApplicationRecord
 
   belongs_to :topic
   belongs_to :user
+
+  validate :max_targets, unless: -> { user.nil? }
```

Comment:
You just want to validate this on create, right? (and maybe on update later)

```suggestion
  validate :max_targets, unless: -> { user.nil? }, on: :create
```

Reply:
Done!